### PR TITLE
Calculate Sen's slope on pre-whitened data in zyp.yuepilon

### PR DESCRIPTION
**Pull request by ateucher**
This brings the behaviour of `zyp.yuepilon` in line with `zyp.zhang` in that the Sen's slope is computed after autocorrelation is removed by the relevant pre-whitening routine. See #2 for discussion

### CODE REVIEW

File: R/zyp.R [hunk 1]
```diff
@@ -185,12 +185,17 @@ zyp.yuepilon <- function(y, x=1:length(y), conf.intervals=TRUE, preserve.range.f
...
   ytnm <- as.numeric(yt[dmap.prime])
+  t.prime <- t.prime[dmap.prime]
 
   # Calculate the Mann-Kendall test for significance on the blended series
-  Kend <- Kendall(t.prime[dmap.prime],ytnm)
+  Kend <- Kendall(t.prime,ytnm)
   tau <- Kend[1]
   Bsig <- Kend[2]
 
+  # Calculate slope on pre-whitened data
```
Comment (jameshiebert) [inline R/zyp.R:195]: If the slope and trend are now calculated here, do they also need to be done above?

**Comment by jameshiebert** [comment]
@AreliaTW Would you mind taking a look at this PR and addressing whether it corrects or changes the intent of the ``zyp.yuepilon`` function?

File: R/zyp.R (hunk 1 above)
Comment (ateucher) [reply R/zyp.R:195]: Yes, as the initial trend is used in the pre-whitening routine in lines 172-188.

**Comment by AreliaTW** [comment]
@ateucher Thanks for your excellent sleuthing. You had mentioned this before by email, but you have explained the issue more eloquently here. I see the potential for your pull request to be a solution. However, I need a few solid hours to think this through. I will dedicate Friday morning to this. Arelia 

@jameshiebert Thanks for passing this along.

**Comment by ateucher** [comment]
Fantastic, thanks so much for looking into this @AreliaTW!

File: R/zyp.R (hunk 1 above)
Comment (AreliaTW) [reply R/zyp.R:195]: @jameshiebert I went through Yue et al. 2002 paper "The influence of autocorrelation on the ability to detect trend in hydrological series." The zyp.yuepilon function in zyp.R follows the proposed procedure for detecting trends in significantly serially correlated hydrological series properly as written. However, the computation of the confidence intervals was not addressed directly in the paper in the same way it was for the zyp.zhang method. Therefore, I see how this was missed when the zyp.yuepilon method was coded up. The changes suggested to the computation of the confidence intervals and trend by @ateucher are correct in my opinion and his pull request should be accepted. Thanks for catching this @ateucher.

**Comment by jameshiebert** [comment]
Perfect! Thanks to both of you!

**Comment by ateucher** [comment]
Thanks @jameshiebert and @AreliaTW! Do you know when you might submit to CRAN?

**Comment by AreliaTW** [comment]
@jameshiebert  and @ateucher - could we wait on submitting it to CRAN? At least till the end of Jan.

**Comment by ateucher** [comment]
Fine with me for sure. Better to be right than fast :)

**Pull request merged**